core/src/main/java/org/apache/calcite/util/BuiltInMethod.java

```diff
@@ -512,8 +512,7 @@ public enum BuiltInMethod {
   SAFE_DIVIDE(SqlFunctions.class, "safeDivide", double.class, double.class),
   SAFE_MULTIPLY(SqlFunctions.class, "safeMultiply", double.class, double.class),
   SAFE_SUBTRACT(SqlFunctions.class, "safeSubtract", double.class, double.class),
-  LOG(SqlFunctions.class, "log", long.class, long.class),
-  LOG2(SqlFunctions.class, "log2", long.class),
+  LOG(SqlFunctions.class, "log", long.class, long.class, int.class),
```

Comment:
Maybe this requires an enum instead of a numeric flag?

Reply:
@tanclary I'm sorry that it took so long to improve it. If you have time, please review this PR.